mlir/lib/Dialect/Affine/IR/AffineOps.cpp

```diff
@@ -4736,6 +4741,14 @@ LogicalResult AffineLinearizeIndexOp::verify() {
 }
 
 OpFoldResult AffineLinearizeIndexOp::fold(FoldAdaptor adaptor) {
+  // No indices linearizes to zero.
+  if (getMultiIndex().empty())
```

Comment:
I think we should disallow this case in the verifier.

Reply:
I think we should allow this case and define it to be 0. It follows naturally from the general idea of linearization.